platform/android/export/export.cpp

```diff
@@ -1455,6 +1483,86 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		}
 	}
 
+	Vector<uint8_t> _resize_launcher_icon(const Ref<Image> &p_source_image, const LauncherIcon p_icon) {
+		Ref<Image> working_image = p_source_image;
+
+		if (p_source_image->get_width() != p_icon.dimensions || p_source_image->get_height() != p_icon.dimensions) {
+			working_image = p_source_image->duplicate();
+			working_image->resize(p_icon.dimensions, p_icon.dimensions, Image::Interpolation::INTERPOLATE_LANCZOS);
+		}
+
+		PoolVector<uint8_t> png_buffer;
+		Error err = PNGDriverCommon::image_to_png(working_image, png_buffer);
+		if (err == OK) {
+			Vector<uint8_t> ret;
+			ret.resize(png_buffer.size());
+			memcpy(ret.ptrw(), png_buffer.read().ptr(), ret.size());
+			return ret;
+		} else {
+			String err_str = String("Failed to convert resized icon (") + p_icon.export_path + ") to png.";
+			WARN_PRINT(err_str.utf8().get_data());
+			return Vector<uint8_t>();
+		}
+	}
+
+	void _copy_icons_to_gradle_project(const Ref<EditorExportPreset> &p_preset) {
```

Comment:
Here also, there's a lot of logic that's duplicated from `export_project`, so extract the common logic instead in a separate function.

Reply:
I don't think this is doable.

In the legacy build, this is how Icons are handled:
1. Template apk is unzipped
2. Every icon file inside the template apk is overwritten with the correct icon data
  a. This is done by checking the filename of the icon file and overwriting it with the appropriately dimensioned icon file
3. The new icon file is saved inside the template apk

In the custom build, this is how icons are handled:
1. For each potential icon dimension, a icon file is generated
2. These icon files are then written to the gradle project directory

I agree that there is a lot of overlapping code, but to consolidate this shared logic into a separate function would not be a cleaner solution because the behavior of the legacy build and the custom build are very different.